### PR TITLE
Fix header casing, bearer redaction, and an unwired test

### PR DESCRIPTION
**Pull request by Phazzie**
Three independent quick wins, each verified against the behavior before the change.

## 1. `authenticateRequest` rejected canonically cased headers

`readHeader` in `api/_lib/middleware/security.ts` looked up `headers?.[name] ?? headers?.[name.toLowerCase()]`. Both call sites already pass lowercase names (`'x-api-key'`, `'authorization'`), so the fallback was dead code — a header bag keyed `X-API-Key` or `Authorization` resolved to `undefined`.

That is exactly the casing the endpoint's own error message tells clients to send:

> `API key required. Include X-API-Key header or Authorization Bearer token`

Observed before the fix:

```
lowercase x-api-key:      {"authenticated":true, ...}
canonical X-API-Key:      {"authenticated":false,"error":{"code":"MISSING_API_KEY",...}}
canonical Authorization:  {"authenticated":false,"error":{"code":"MISSING_API_KEY",...}}
```

Header names are case-insensitive per RFC 7230 §3.2. `findHeaderValue` now matches on the lowercased key, keeping the direct lookup as the fast path since Node's `IncomingMessage.headers` is already lowercased. All three cases authenticate after the fix.

Scope note: on Vercel the request goes through Node's lowercasing, so production traffic was not affected. What was affected is every other caller — hand-built header objects, fetch-style adapters, and test fixtures — and the function's contract, which claimed a case-insensitive lookup it did not perform.

## 2. `redactSensitiveTextTokens` corrupted words containing "bearer"

The scanner in `shared/sensitiveTextRedaction.ts` treated any occurrence of the substring `bearer` as the auth scheme, with no boundary check before it. An ordinary word that merely contains it was rewritten to the scheme's casing, and the *following* word was swallowed as if it were the credential:

```
"forbearer patterns are fine"  ->  "forBearer [REDACTED] are fine"
```

The trailing boundary was already handled (`"Bearerish token here"` was correctly left alone) — only the leading one was missing. `hasBearerBoundaryBefore` supplies it.

**Correction made during review.** The first version of that guard reused `isBearerTokenChar`, for symmetry with the `hasApiTokenBoundaryBefore` check on the API-key path. That symmetry was wrong and introduced a leak that did not exist before: the token grammar counts `=`, `/` and `+` as word characters, so a credential in key/value form passed through in the clear.

```
"Authorization=Bearer abc123def456"  ->  "Authorization=Bearer abc123def456"   // leaked
```

The two guards are not analogous — the API-key one delimits a *token*, this one delimits a *word* — and only letters and digits can put `bearer` inside a longer word. The guard now uses `isAsciiLetterOrDigit`, and the comment on it records why the narrower class is deliberate. Final behavior:

```
"Authorization=Bearer abc123def456"   ->  "Authorization=Bearer [REDACTED]"
"Authorization: Bearer abc123def456"  ->  "Authorization: Bearer [REDACTED]"
"auth/Bearer abc123def456"            ->  "auth/Bearer [REDACTED]"
"forbearer patterns are fine"         ->  "forbearer patterns are fine"
"torchbearer patterns are fine"       ->  "torchbearer patterns are fine"
```

One tradeoff left deliberately in place: `"the bearer of bad news"` still redacts, because `bearer` there does start a word. Tightening further (requiring an `Authorization:` context) would risk missing real tokens, and over-redaction is the safe direction for a redactor.

## 3. `story-lab-blueprint-parser.test.ts` never ran

The file had no npm script and was absent from `test:all`, so the Story Lab request-validation boundary — creature/tone/spicy-level/word-budget/batch-size parsing, theme and heat-contract shape checks — had zero CI coverage. It passes as written; wired in as `test:story-lab-blueprint-parser` and added to the `test:all` chain.

Three other test files are also unwired (`frontend-streaming`, `verify-ai-fixes`, `story-service.test.mjs`). I left them out: `frontend-streaming` fails to resolve a module and `verify-ai-fixes` prints notes rather than asserting, so wiring them in is a repair job rather than a quick win.

## Testing

- Full `npm run test:all` — green, including the newly wired blueprint parser test.
- API typecheck (`tsc --noEmit` over `api/**/*.ts`, matching `preflight.sh`) — clean.
- Regression tests added for (1) in `tests/api-key-auth.test.ts` and (2) in `tests/log-redaction.test.ts`; both fail against the pre-fix code. The redaction test now also loops over the delimiters that are themselves valid bearer-token characters (`=`, `:`, `/`, `+`, `"`, `,`, `(`) and asserts the credential is gone for each.
- Slice recorded in `PR70_RECOVERY_CHANGELOG.md` per the documentation update map in `AGENTS.md`.

## Not addressed here

`authenticateRequest` and `checkRateLimit` are never called by any API handler — only by tests. The auth middleware is effectively unwired from live routes. That is a real gap, but connecting it is a behavior change that needs its own decision, so it is untouched by this PR.

Note for reviewers: `node_modules` is tracked in this repo. Installing dependencies to run the suite modified those tracked files; they were restored so this diff is source-only.

## Summary by Sourcery

Correct header authentication and bearer redaction edge cases, and include the Story Lab blueprint parser in CI test coverage.

Bug Fixes:
- Make authentication header lookup case-insensitive so canonically cased API key and Authorization headers are accepted.
- Prevent bearer-token redaction from corrupting ordinary words while continuing to redact credentials across supported delimiters.

CI:
- Wire the Story Lab blueprint parser test into its own npm script and the full test suite.

Documentation:
- Record the header, redaction, and test-coverage changes in the recovery changelog.

Tests:
- Add regression coverage for canonical header casing, bearer-containing words, and delimiter-prefixed bearer credentials.

### CODE REVIEW

File: shared/sensitiveTextRedaction.ts [hunk 1]
```diff
@@ -22,6 +22,16 @@ function redactBearerTokens(value: string): string {
 
     redacted += value.slice(index, found);
 
+    // `bearer` only introduces a credential when it starts a word. Without this
+    // guard an ordinary word that merely contains the substring (`forbearer`,
+    // `overbearing`) is rewritten to the scheme's casing and the following word
+    // is swallowed as if it were a token.
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply shared/sensitiveTextRedaction.ts]: Good catch — `overbearing` contains `bear`, not `bearer`, so it was never affected by the missing boundary check and did not illustrate the bug at all.

Swapped it for `torchbearer` in both the explanatory comment and the regression test in `tests/log-redaction.test.ts`, which had the same misleading example. Comment and test wording only; no behavior change.

Fixed in a9c2e46.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: shared/sensitiveTextRedaction.ts [hunk 2]
```diff
@@ -22,6 +22,16 @@ function redactBearerTokens(value: string): string {
...
+    if (!hasBearerBoundaryBefore(value, found)) {
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply shared/sensitiveTextRedaction.ts]: Confirmed and fixed — this was a real leak I introduced. Reproduced before the fix:

```
"Authorization=Bearer abc123def456"  ->  "Authorization=Bearer abc123def456"   // leaked
"auth/Bearer abc123def456"           ->  "auth/Bearer abc123def456"            // leaked
"x+Bearer abc123def456"              ->  "x+Bearer abc123def456"               // leaked
```

The root mistake was reaching for `isBearerTokenChar` for symmetry with `hasApiTokenBoundaryBefore`. The two guards are not symmetric: the API-key one delimits a *token*, this one delimits a *word*. Only letters and digits can put `bearer` inside a longer word, so `hasBearerBoundaryBefore` now uses `isAsciiLetterOrDigit`. After the fix:

```
"Authorization=Bearer abc123def456"  ->  "Authorization=Bearer [REDACTED]"
"auth/Bearer abc123def456"           ->  "auth/Bearer [REDACTED]"
"x+Bearer abc123def456"              ->  "x+Bearer [REDACTED]"
"forbearer patterns are fine"        ->  "forbearer patterns are fine"
```

Added the negative redaction cases you asked for — `tests/log-redaction.test.ts` now loops over `=`, `:`, `/`, `+`, `"`, `,` and `(` and asserts the credential is gone for each. The comment on the guard records why it deliberately uses a narrower class than the token grammar, so the next reader doesn't re-introduce the symmetry.

Fixed in b3444d6.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: api/_lib/middleware/security.ts [hunk 3]
```diff
@@ -91,13 +91,42 @@ export async function authenticateRequest(req: AuthenticatedRequest): Promise<Au
...
+ * this function with header bags that have not always been through Node's
+ * lowercasing `IncomingMessage.headers` — a hand-built object, a fetch-style
+ * adapter, or a test fixture may carry the canonical `X-API-Key` /
+ * `Authorization` casing that the documentation and the MISSING_API_KEY
+ * message tell clients to send. Match on the lowercased key so every casing
+ * resolves to the same header.
  */
 function readHeader(headers: any, name: string): string | undefined {
-  const raw = headers?.[name] ?? headers?.[name.toLowerCase()];
+  const raw = findHeaderValue(headers, name);
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply api/_lib/middleware/security.ts]: Agreed — the documentation update map in AGENTS.md names `PR70_RECOVERY_CHANGELOG.md` as the default running log for any slice that changes behavior, validation evidence, or review follow-up, and this slice does all three.

Added an entry in b3444d6 following the file's existing Actions / Self-review / Validation structure. It records the three fixes, the correction to my own first attempt at the bearer boundary (the leak you caught in the sibling thread), the validation results, and two explicit non-claims — that `authenticateRequest` and `checkRateLimit` are still called only from tests and remain unwired from live routes, and that three other test files stay unwired because they need repair rather than wiring.

---
_Generated by [Claude Code](https://claude.ai/code)_
